.pre-commit-config.yaml

```diff
@@ -0,0 +1,31 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
+  # https://pycqa.github.io/isort/docs/configuration/pre-commit.html
+  - repo: https://github.com/pycqa/isort
+    rev: 5.12.0
+    hooks:
+      - id: isort
+        name: isort (python)
+    # https://black.readthedocs.io/en/stable/integrations/source_version_control.html
+  - repo: https://github.com/psf/black
+    rev: 23.1.0
```

Comment:
Is this going to be sustainable to have version numbers in this file?  They may not be consistent with what is running on CI.

Reply:
Hi Chris, good question. I was wondering the same thing. Having the version numbers does seem potentially problematic, but it's apparently [how the tool is intended to be used](https://[pre-commit](https://pre-commit.com/#using-the-latest-version-for-a-repository).com/#using-the-latest-version-for-a-repository). More discussion [here](https://github.com/psf/black/issues/420) re: black from one of pre-commit's maintainers. Maybe we can simply add a statement in the developer docs to let people know that the versions might be different, and that they can update to the latest hooks via the `[pre-commit autoupdate]`? @w-bonelli maybe we could also add `[pre-commit autoupdate]` to the release workflow somehow, so that every time a release is made, it gets run (and `.pre-commit-config.yaml` gets updated? 

As things are now, wouldn't anyone with an older version of black or isort on their local machine be in the same situation?

Reply:
> As things are now, wouldn't anyone with an older version of black or isort on their local machine be in the same situation?

Yes we [ran into this](https://github.com/modflowpy/flopy/issues/1697#issuecomment-1414465812) just last week.

Adding `pre-commit autoupdate` to the release workflow should be straightforward, it could run before the post-release PR updating `develop` from `master` is opened (we already [lint and update version strings](https://github.com/modflowpy/flopy/blob/develop/.github/workflows/release.yml#L364) at this point). Seems like this would generally avoid having to manually autoupdate locally, if I understand correctly? (agree it's probably still good to have a note in the dev docs)

Reply:
I've added a note about the static versions and `pre-commit autoupdate` to the `DEVELOPER.md` doc.